File: app/src/main/java/com/example/monappliimmo/InterventionDetailActivity.java
package com.example.monappliimmo;

import android.os.Bundle;
import android.view.MenuItem;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

public class InterventionDetailActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_intervention_detail);

        // 1. Récupérer les composants UI
        TextView tvType = findViewById(R.id.detailType);
        TextView tvDate = findViewById(R.id.detailDate);
        TextView tvDesc = findViewById(R.id.detailDesc);

        // 2. Activer la flèche de retour (déplacé ici)
        if (getSupportActionBar() != null) {
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
            getSupportActionBar().setDisplayShowHomeEnabled(true);
            getSupportActionBar().setTitle("Détail intervention");
        }

        // 3. Récupérer l'objet Intervention
        Intervention inter = (Intervention) getIntent().getSerializableExtra("intervention_selectionnee");

        if (inter != null) {
            tvType.setText(inter.getType());
            tvDesc.setText(inter.getDescription());

            if (inter.getDatePrevue() != null) {
                tvDate.setText(formatteDateEurope(inter.getDatePrevue()));
            } else {
                tvDate.setText("Non planifiée");
            }
        }
    }

    private String formatteDateEurope(String dateBrute) {
        try {
            SimpleDateFormat sdfSource = new SimpleDateFormat("yyyy-MM-dd", Locale.getDefault());
            SimpleDateFormat sdfDestination = new SimpleDateFormat("dd/MM/yyyy", Locale.FRANCE);
            Date date = sdfSource.parse(dateBrute);
            return sdfDestination.format(date);
        } catch (ParseException e) {
            return dateBrute;
        }
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            finish();
            return true;
        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/com/example/monappliimmo/InterventionDetailActivity.kt
package com.example.monappliimmo

import android.os.Bundle
import android.view.MenuItem
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.Locale

class InterventionDetailActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_intervention_detail)

        // 1. Récupérer les composants UI
        val tvType = findViewById<TextView>(R.id.detailType)
        val tvDate = findViewById<TextView>(R.id.detailDate)
        val tvDesc = findViewById<TextView>(R.id.detailDesc)
        val tvAppart = findViewById<TextView>(R.id.detailAppartement) // Ajout de l'appartement
        val btnRetour = findViewById<Button>(R.id.btnRetour) // Ajout du bouton

        // 2. Activer la flèche de retour dans la barre du haut
        supportActionBar?.apply {
            setDisplayHomeAsUpEnabled(true)
            setDisplayShowHomeEnabled(true)
            title = "Détail intervention"
        }

        // 3. Récupérer l'objet Intervention
        val inter = intent.getSerializableExtra("intervention_selectionnee") as? Intervention

        inter?.let {
            tvType.text = it.type
            tvDesc.text = it.description

            // Affichage de l'appartement lié
            tvAppart.text = it.appartement?.numero?.toString() ?: "Appartement non spécifié"

            // Formatage de la date
            tvDate.text = it.datePrevue?.let { date -> formatteDateEurope(date) } ?: "Non planifiée"
        }

        // 4. Action du bouton "Retour à la liste" en bas de l'écran
        btnRetour.setOnClickListener {
            finish() // Ferme l'activité et revient à la liste
        }
    }

    private fun formatteDateEurope(dateBrute: String): String {
        return try {
            val sdfSource = SimpleDateFormat("yyyy-MM-dd", Locale.getDefault())
            val sdfDestination = SimpleDateFormat("dd/MM/yyyy", Locale.FRANCE)
            val date = sdfSource.parse(dateBrute)
            sdfDestination.format(date!!)
        } catch (e: Exception) {
            dateBrute
        }
    }

    // Gestion du clic sur la flèche de retour (en haut à gauche)
    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            finish()
            return true
        }
        return super.onOptionsItemSelected(item)
    }
}